src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -44,6 +44,9 @@ export type HostConfig<T, P, I, C> = {
   commitUpdate(instance : I, oldProps : P, newProps : P, children : HostChildren<I>) : void,
   deleteInstance(instance : I) : void,
 
+  createTextInstance(text : string) : TI,
+  commitTextUpdate(textInstance : TI, oldText : string, newText : string) : void,
```

Comment:
An alternative approach would overload the existing `createInstance` and `commitUpdate` to accept a `string` and return a `Text` node. This might seem like it causes unnecessary branching trouble there, but if you think about it, all other instances tend to be more specific types than "View" or "Element". They're all subclasses so you need to branch on them sooner or later anyway.

There isn't necessarily any need to special case Text components here.

It is needed in the internals for reconciliation purposes since string children are special. However, they're not needed here.

You could argue that since they're already special cased by having their own fiber, it allows early branching instead of branching in the switch statement and then branching again inside the host environment adapter.